impala/tests/test_sqlalchemy.py

```diff
@@ -52,3 +52,32 @@ def test_sqlalchemy_compilation():
                 "TBLPROPERTIES ('kudu.table_name' = 'my_kudu_table', "
                 "'kudu.master_addresses' = 'kudu-master.example.com:7051')\n\n")
     assert expected == observed
+
+def test_sqlalchemy_impala4_compilation():
+    def create_test_engine():
+        return create_engine('impala4://{0}:{1}'.format(TEST_ENV.host, TEST_ENV.port))
+
+    engine = create_test_engine()
+    metadata = MetaData(engine)
+    # TODO: add other types to this table (e.g., functional.all_types)
+    mytable = Table("mytable",
+                    metadata,
+                    Column('col1', STRING),
+                    Column('col2', TINYINT),
+                    Column('col3', INT),
+                    Column('col4', DOUBLE),
+                    Column('col5', DATE),
+                    Column('col6', VARCHAR(10)),
+                    impala_partition_by='HASH PARTITIONS 16',
+                    impala_stored_as='KUDU',
+                    impala_table_properties={
+                        'kudu.table_name': 'my_kudu_table',
+                        'kudu.master_addresses': 'kudu-master.example.com:7051'
+                    })
+    observed = str(CreateTable(mytable, bind=engine))
+    expected = ('\nCREATE TABLE mytable (\n\tcol1 STRING, \n\tcol2 TINYINT, '
```

Comment:
This  test seems misleading to me, as this CREATE TABLE ... STORED AS KUDU would be rejected as there is no primary key column, which is mandatory in Kudu. Can you rewrite it to a non-Kudu, e.g. Parquet table? It should use impala_partitioned_by instead of impala_partition_by

Reply:
Done